test/e2e/deployment.go

```diff
@@ -1013,3 +1016,79 @@ func testScalePausedDeployment(f *framework.Framework) {
 		Expect(err).NotTo(HaveOccurred())
 	}
 }
+
+func testScaledRolloutDeployment(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := adapter.FromUnversionedClient(f.Client)
+
+	podLabels := map[string]string{"name": nginxImageName}
+	replicas := int32(10)
+
+	// Create a nginx deployment.
+	deploymentName := "nginx"
+	d := newDeployment(deploymentName, replicas, podLabels, nginxImageName, nginxImage, extensions.RollingUpdateDeploymentStrategyType, nil)
+	d.Spec.Strategy.RollingUpdate = new(extensions.RollingUpdateDeployment)
+	d.Spec.Strategy.RollingUpdate.MaxSurge = intstr.FromInt(3)
+	d.Spec.Strategy.RollingUpdate.MaxUnavailable = intstr.FromInt(2)
+	framework.Logf("Creating deployment %q", deploymentName)
+	_, err := c.Extensions().Deployments(ns).Create(d)
+	Expect(err).NotTo(HaveOccurred())
+	defer stopDeployment(c, f.Client, ns, deploymentName)
+
+	// Check that deployment is created fine.
+	deployment, err := c.Extensions().Deployments(ns).Get(deploymentName)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = framework.WaitForObservedDeployment(c, ns, deploymentName, deployment.Generation)
+	Expect(err).NotTo(HaveOccurred())
+
+	first, err := deploymentutil.GetNewReplicaSet(deployment, c)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Update the deployment with a broken image so that the new replica set will be blocked.
+	framework.Logf("Updating deployment %q with a broken image", deploymentName)
```

Comment:
`By` is more suitable here. 

Logf creates something like this 
```
Jul 13 08:37:53.992: INFO: Updating deployment "nginx" with a broken image
```
whereas By creates this
```
STEP: Updating deployment "nginx" with a broken image
```

I know we've been using Logf heavily in deployment e2e test, but let's not copy that pattern in this new test.